lib/protocol/protocol.js

```diff
@@ -352,7 +354,7 @@ function buildHandler (app, method, path, spec, driver, isSessCmd) {
       if (isErrorType(err, errors.ProxyRequestError)) {
         protocolLogger.error(`Encountered internal error running command:  ${JSON.stringify(err)} ${err.stack}`);
         actualErr = err.getActualError();
-      } else if (!isW3C && (!(isErrorType(err, ProtocolError) || isErrorType(err, errors.BadParametersError)))) {
+      } else if (!isW3C && isUnknownError(err)) {
```

Comment:
I still don't quite understand why `!isW3C` is necessary here

Reply:
perhaps @dpgraham can explain

Reply:
I have no idea. It was there, so I'm not ready to remove.

Reply:
The `getResponseForW3CError` that's called afterwards covers the conversions so it was unnecessary.

Reply:
lets add an explanation comment here then